sql/src/main/java/io/crate/execution/dml/upsert/CheckConstraints.java

```diff
@@ -73,6 +86,22 @@ public void validate(T values) {
                 throw new IllegalArgumentException("\"" + notNullColumns.get(i) + "\" must not be null");
             }
         }
+        for (Map.Entry<String, Tuple<Input<?>, String>> checkEntry : checkConstraints.entrySet()) {
+            Input<?> checkInput = checkEntry.getValue().v1();
+            Boolean value = (Boolean) checkInput.value();
+            if (null == value) {
+                continue;
+            }
+            if (!value.booleanValue()) {
+                String checkName = checkEntry.getKey();
+                String checkExpr = checkEntry.getValue().v2();
+                Object offendingValues = values instanceof Doc ? ((Doc) values).getSource() : values;
```

Comment:
Can we get rid of the instanceof here? In this case I think the `CheckConstraints` shouldn't have any knowledge about what `T` instances are possible.

Reply:
Doc does not have a toString(), so in principle I could add it and print out only the value of sources... which to me is incorrect. Maybe T could extend `WithRepresentableValues` ?